src/built-in-macros.ls

```diff
@@ -244,20 +244,49 @@ contents =
 
   \. : do
 
-    is-computed-property = (ast-node) ->
-      switch ast-node.type
-      | \Identifier => false
-      | otherwise => true
+    # Terminology:
+    #
+    #  - a _computed_ member expression involves square brackets, like `a[b]`.
+    #  - a _non-computed_ member expression involves a dot like `a.b`.
+
+    # Assumptions:
+    #
+    # - The estree spec requires a non-computed member expression's property to
+    #   be an Identifer node.
+    # - There's no reason we'd ever want to write `a["b"]`, as that is
+    #   completely equivalent to `a.b` and easier to read.
+
+    make-member-expression = ({compile}:env, object, property) ->
+
+      var computed
+      var property-compiled
+
+      # Compile the property to an estree node, so we can decide what to do
+      # with it
+      property-as-estree = compile property
+
+      # Ensure that a string literal is converted to an identifer and compiled
+      # into a non-computed member expression
+      if property-as-estree.type is \Literal
+      and typeof property-as-estree.value is \string
```

Comment:
Right, that's the problem.  Typing `(. a "foo-bar")` (which *should* be valid) will produce an AST error. There's no reason for that to error out, particularly since the `a.b` transformation (instead of `a["b"]`) is only done for perf anyway.  Instead, `(. a "foo-bar")` should be detected at this point, and instead desugar to `a["foo-bar"]` (in other words, follow the other branch).